packages/block-library/src/query/edit/pattern-selection.js

```diff
@@ -43,7 +43,16 @@ export function useBlockPatterns( clientId, attributes ) {
 		clientId,
 		attributes
 	);
-	return usePatterns( clientId, blockNameForPatterns );
+	const allPatterns = usePatterns( clientId, blockNameForPatterns );
+	// Filter out any patterns that don't have Query as their root block
+	// so that a Query block is always replaced by another Query block.
+	const rootBlockPatterns = useMemo( () => {
+		return allPatterns.filter( ( pattern ) => {
+			return pattern.blocks?.[ 0 ]?.name === blockNameForPatterns;
+		} );
+	}, [ allPatterns, blockNameForPatterns ] );
```

Comment:
Hey folks! 👋 

I’m maintaining Sensei LMS and noticed that one of our [blocks](https://github.com/Automattic/sensei/blob/trunk/assets/blocks/course-list-block/index.js#L34) (block variation of core/query) [stopped showing the “Choose” patterns button](https://github.com/Automattic/sensei/issues/7873), and it looks like this change is related. For context, the custom patterns are registered [here](https://github.com/Automattic/sensei/blob/trunk/includes/block-patterns/course-list/class-sensei-course-list-block-patterns.php). Could you give me a hand with finding a solution?

Reply:
@m1r0 would you be able to create a new issue for this and then link to it from here?